lib/rubocop/cop/rails/has_many_or_has_one_dependent.rb

```diff
@@ -39,17 +39,33 @@ class HasManyOrHasOneDependent < Cop
           (pair (sym :through) !nil)
         PATTERN
 
+        def_node_matcher :with_options_block?, <<-PATTERN
+          (block
+            (send nil? :with_options
+              (hash $...))
+            (args) ...)
+        PATTERN
+
         def on_send(node)
-          unless is_has_many_or_has_one_without_options?(node)
-            pairs = is_has_many_or_has_one_with_options?(node)
-            return unless pairs
-            return if pairs.any? do |pair|
-              has_dependent?(pair) || has_through?(pair)
-            end
+          if !is_has_many_or_has_one_without_options?(node)
```

Comment:
I would scissor the `is` prefix from the two matchers, since it would be an offense in a regular method definition. (Perhaps we should transpose the method naming cops to an internal investigation cop that checks dynamically generated methods.) 🙂

Reply:
I fixed them and squashed.

> Perhaps we should transpose the method naming cops to an internal investigation cop that checks dynamically generated methods.

Sure! It is not the scope of this PR, but that is useful if we can catch it with a rule like `Naming/PredicateName` cop :memo: